### PR TITLE
V2.2.1 Release -- patches to docs and build

### PR DESCRIPTION
**Pull request by rstoneback**
# Description

Addresses # (issue)

Distribution settings bug. See pull #491 

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Local install from an sdist generated distribution and visual verification that Instrument modules pysat_testing* are present. 

# Checklist:

- [ ] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by aburrell** [comment]
The files look good.  Not sure how to test this, though.  Ideas, @jklenzing ?

**Comment by rstoneback** [comment]
https://packaging.python.org/guides/using-testpypi/

There is a test pypi server that we can use and that pip install can download from.

**Comment by rstoneback** [comment]
`pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple pysat  `

```
In [1]: import pysat

In [2]: pysat.__version__
Out[2]: '2.2.2'

```

**Comment by aburrell** [comment]
That's a good step, that I normally do for making a release of my own packages, but that doesn't really test the behaviour of the package.  What are we looking for?  How do we make sure that we're not missing another obvious bug here?

I suppose what we really need at this point are integration tests, but that's for 3.X...

**Comment by jklenzing** [comment]
We could test a branch of one of the dependent packages using the test pypi server to ensure that the test instruments are behaving correctly.

**Comment by aburrell** [comment]
Yeah, but which one?  I don't think any package has good unit tests yet :/

**Comment by jklenzing** [comment]
pysatSeasons should work.  It will also be a good test of backward compatibility with older versions of pandas, etc

**Comment by jklenzing** [comment]
Running a test now.  so far, so good...

**Comment by jklenzing** [comment]
Updated travis file here: https://github.com/pysat/pysatSeasons/blob/test_py222/.travis.yml
Tests passed here:  https://travis-ci.org/github/pysat/pysatSeasons/builds/712692566

- verified in the log that pysat 2.2.2 is installed.  It only uses `pysat_testing` and `pysat_testing2d`, and there's no tests using custom functions, but it looks good.  
- pysatSeasons coverage is 97%, and should be identical to the ssnl routines in 2.2.2.

**Comment by jklenzing** [comment]
And for comparison, here's a version where we just use `pip install pysat` to get 2.1.  Every test fails, since every tests needs a test instrument.

https://travis-ci.org/github/pysat/pysatSeasons/builds/712695008

**Comment by jklenzing** [comment]
Path forward:
- remove prune tests
- run all subpackages installing from pypi
Results:
- [ ] pysatMadrigal
- [ ] pysatMissions
- [ ] pysatModels
- [x] pysatSeasons

File: MANIFEST.in [hunk 1]
```diff
@@ -6,9 +6,7 @@ include *.md
 include description.txt
 include LICENSE
 include pysat/version.txt
-prune pysat/tests
 prune docs
 prune demo
-recursive-exclude pysat/instruments pysat_testing*.py
 exclude *.pdf
 exclude *.png
```
Comment (aburrell) [inline MANIFEST.in:12]: This wants a new line

**Comment by jklenzing** [comment]
Still poking at the tests.  One thought came up post meeting: do we want to include the rename "default branch to main" as part of the patch release?  We can do the swap over regardless, but it does require a few updates to documentation.

**Comment by aburrell** [comment]
Might as well change it over now.  This will be the last bit of documentation for 2.X (hopefully), so nicer to have it change now.

**Comment by jklenzing** [comment]
#494 created as draft.   I've also added a main branch, which is a copy of master.  Once the tests pass, we can pull everything together.

**Comment by rstoneback** [comment]
Thanks @jklenzing!

**Comment by jklenzing** [comment]
Changed the base and tests are looking good so far.  This is ready to merge once the tests finish

**Comment by jklenzing** [comment]
Just noticed there was no official release of 2.2.1.  Are we renaming or just skipping that one?

**Comment by rstoneback** [comment]
My initial thinking was to just skip the official release of 2.2.1. Since it was never released though (didn’t go to pip or come out on GitHub) we could/should call this one v2.2.1.

On Aug 3, 2020, at 10:51 AM, Jeff Klenzing <notifications@github.com> wrote:

﻿

Just noticed there was no official release of 2.2.1. Are we renaming or just skipping that one?

**Comment by jklenzing** [comment]
OK.  I can patch the version number and changelog in this one before merging.  @aburrell, any other thoughts?

**Comment by aburrell** [comment]
Nope.  I have no more thoughts.

**Comment by jklenzing** [comment]
OK.  fixed changelog and version number, as well as the newline comment.  Waiting on tests...

**Comment by rstoneback** [comment]
Thanks @jklenzing and @aburrell!

**Comment by aburrell** [comment]
Waiting on the pip release of this version for https://github.com/aburrell/ocbpy/pull/74.

**Pull request merged**